### PR TITLE
add p3_sk_test and fix the bfb bug in nightly test

### PR DESCRIPTION
**Pull request by xyuan**
add p3_sk_test and fix the bfb bug in nightly test

### CODE REVIEW

**Comment by xyuan** [comment]
Permultter:
./cime/scripts/create_test ERS_Ln90.ne30pg2_ne30pg2.F2010-SCREAMv1.pm-gpu_gnugpu.scream-small_kernels (passed)
Frontier:
./cime/scripts/create_test ERS_Ln90.ne30pg2_ne30pg2.F2010-SCREAMv1.frontier-scream-gpu_crayclang-scream.scream-small_kernels (passed)
Summit:
./cime/scripts/create_test ERS_Ln90.ne30pg2_ne30pg2.F2010-SCREAMv1.frontier-summit_gnugpu.scream-small_kernels (passed)

**Comment by yuenxq** [comment]
weaver is hard to understand the differences without login the system, 

bash-4.4$ ctest
Test project /ccs/home/yuanx/scream_p3_sk/components/eamxx/ctest-build/full_debug/src/physics/p3
    Start 1: p3_tests_omp1
1/9 Test #1: p3_tests_omp1 ....................   Passed   28.61 sec
    Start 2: p3_tests_fail_omp1
2/9 Test #2: p3_tests_fail_omp1 ...............   Passed    6.92 sec
    Start 3: p3_sk_tests_omp1
3/9 Test #3: p3_sk_tests_omp1 .................   Passed   23.98 sec
    Start 4: p3_sk_tests_fail_omp1
4/9 Test #4: p3_sk_tests_fail_omp1 ............   Passed    5.15 sec
    Start 5: p3_run_and_cmp_cxx
5/9 Test #5: p3_run_and_cmp_cxx ...............***Failed    1.13 sec
    Start 6: p3_run_and_cmp_f90
6/9 Test #6: p3_run_and_cmp_f90 ...............***Failed    1.11 sec
    Start 7: p3_run_and_cmp_cxx_fail
7/9 Test #7: p3_run_and_cmp_cxx_fail ..........   Passed    1.11 sec
    Start 8: p3_baseline_f90_fake
8/9 Test #8: p3_baseline_f90_fake .............***Not Run (Disabled)   0.00 sec
    Start 9: p3_baseline_cxx_fake
9/9 Test #9: p3_baseline_cxx_fake .............***Not Run (Disabled)   0.00 sec

71% tests passed, 2 tests failed out of 7

**Comment by xyuan** [comment]
using the same version of CUDA on ANL's machine, 

Currently Loaded Modulefiles:
 1) cuda/11.8.0   2) gcc/11.1.0   3) openmpi/4.1.1-gcc   4) cmake/3.27.2  

xyuan@gpu06:/gpfs/jlse-fs0/projects/climate/xyuan/scream/components/eamxx/build/src/physics/p3> ctest
Test project /gpfs/jlse-fs0/projects/climate/xyuan/scream/components/eamxx/build/src/physics/p3
    Start 1: p3_tests_omp1
1/9 Test #1: p3_tests_omp1 ....................   Passed    1.71 sec
    Start 2: p3_tests_fail_omp1
2/9 Test #2: p3_tests_fail_omp1 ...............   Passed    0.77 sec
    Start 3: p3_sk_tests_omp1
3/9 Test #3: p3_sk_tests_omp1 .................   Passed    1.52 sec
    Start 4: p3_sk_tests_fail_omp1
4/9 Test #4: p3_sk_tests_fail_omp1 ............   Passed    0.85 sec
    Start 5: p3_run_and_cmp_cxx
5/9 Test #5: p3_run_and_cmp_cxx ...............***Failed    2.10 sec
    Start 6: p3_run_and_cmp_f90
6/9 Test #6: p3_run_and_cmp_f90 ...............***Failed    2.11 sec
    Start 7: p3_run_and_cmp_cxx_fail
7/9 Test #7: p3_run_and_cmp_cxx_fail ..........***Failed    2.13 sec
    Start 8: p3_baseline_f90_fake
8/9 Test #8: p3_baseline_f90_fake .............***Not Run (Disabled)   0.00 sec
    Start 9: p3_baseline_cxx_fake
9/9 Test #9: p3_baseline_cxx_fake .............***Not Run (Disabled)   0.00 sec

57% tests passed, 3 tests failed out of 7

**Comment by bartgol** [comment]
I can take a look at weaver.

**Comment by yuenxq** [comment]
> I can take a look at weaver.
> 
> @xyuan looks like replacing the pfor with three deep copies fixes the segfault. If/when you change that, can you also fix the typo `lapvap`->`latvap`?

yeah, I fixed it in the new push

File: components/eamxx/src/physics/p3/impl/p3_get_latent_heat_impl.hpp [hunk 1]
```diff
@@ -12,19 +12,19 @@ ::get_latent_heat(const Int& nj, const Int& nk, view_2d<Spack>& v, view_2d<Spack
...
-      v(i,k) = lapvap;
-      s(i,k) = lapvap + latice;
-      f(i,k) = latice;
-    });
-  });
+  Kokkos::deep_copy(v, latvap);
+  Kokkos::deep_copy(s, latvap + latice);
+  Kokkos::deep_copy(f, latice);
 
+//  Kokkos::parallel_for(
```
Comment (bartgol) [inline components/eamxx/src/physics/p3/impl/p3_get_latent_heat_impl.hpp:22]: Let's get rid of commented code.

**Review by bartgol — APPROVED** [review]

**Pull request merged**